src/main/java/graphql/execution/ValuesResolverConversion.java

```diff
@@ -283,33 +396,69 @@ static CoercedVariables externalValueToInternalValueForVariables(GraphQLSchema s
      * Performs validation too
      */
     @SuppressWarnings("unchecked")
-    static Object externalValueToInternalValueImpl(GraphqlFieldVisibility fieldVisibility,
-                                                   GraphQLType graphQLType,
-                                                   Object value,
-                                                   GraphQLContext graphqlContext,
-                                                   Locale locale) throws NonNullableValueCoercedAsNullException, CoercingParseValueException {
+    static Object externalValueToInternalValueImpl(
+            InputInterceptor inputInterceptor,
+            GraphqlFieldVisibility fieldVisibility,
+            GraphQLInputType graphQLType,
+            Object originalValue,
+            GraphQLContext graphqlContext,
+            Locale locale
+    ) throws NonNullableValueCoercedAsNullException, CoercingParseValueException {
         if (isNonNull(graphQLType)) {
-            Object returnValue =
-                    externalValueToInternalValueImpl(fieldVisibility, unwrapOne(graphQLType), value, graphqlContext, locale);
+            Object returnValue = externalValueToInternalValueImpl(
+                    inputInterceptor,
+                    fieldVisibility,
+                    unwrapOneAs(graphQLType),
+                    originalValue,
+                    graphqlContext,
+                    locale);
             if (returnValue == null) {
                 throw new NonNullableValueCoercedAsNullException(graphQLType);
             }
             return returnValue;
         }
-
```

Comment:
PSSSSSTT!  DOWN HERE!!

This is the main change.  During variable coercion (and only variables - not literals) the callback is made if present BEFORE the value is presented to the scalar / enum coercions

So you can see what is coming in raw and possible change it or just log it say if its unexpected.

You CANT intercept the value AFTER its been co-erced.  I didnt add that capability because I don't see a need for it